src/System.Net.Http/tests/UnitTests/HPack/HPackIntegerTest.cs

```diff
@@ -84,7 +85,7 @@ public void HPack_Integer_TooLarge(byte[] encoded)
             {
                 var dec = new IntegerDecoder();
 
-                if (!dec.StartDecode(encoded[0], 7))
+                if (!dec.StartDecode((byte)(encoded[0] & 0x7F), 7))
```

Comment:
Shouldn't prefixLength here be 1, not 7?

Since you are in this code, can you add the assert to StartDecode that I suggested previously? That would have caught this.

Reply:
In this case `prefixLength` is the number of integer bits in the "prefix byte", so 7 is correct. I can see how the name is confusing, perhaps we should change it.

Good call on the `assert`.

Reply:
Yeah, I see now. This is super confusing. `prefixLength` isn't the length of the prefix, as you might expect. Can we change the name, at least?

Reply:
The name "prefix" is actually standard language that makes sense in the context of the HPACK spec:

> An integer is represented in two parts: a prefix that fills the current octet and an optional list of octets that are used if the integer value does not fit within the prefix.

So I can see why the original code used that. I'm having trouble thinking of a better name and am leaning toward simply improving the comments on the method instead. Can you think of something better?

Reply:
Huh. Well, given the usage in the RFC, it seems reasonable to me to leave as is and improve the comment, as you've done. Thanks for following up here.